Fix building with recent Node

rrthomas opened the pull request:
Two fixes (one to the build system, one to the binding code), and one cosmetic improvement, to remove a configure option for pcre2 that has no effect.

If you are still interested in maintaining this package, it would be good to get in touch with @Kamil93 with an updated version of your PR (I am happy to try!), to see if he wants to update his npm package. Otherwise perhaps you would like to publish one?

rrthomas committed:
Update for Node 4.x and later

Object::Set requires a Local<Context> argument.

rrthomas committed:
compile: remove obsolete --disable-stack-for-recursion option to configure

rrthomas committed:
Remove src/*.c from deps/pcre2/.gitignore

This stops these files being checked out by node-gyp, and the library then
cannot be built.

ghost commented:
I am not interested in developing this module further and have no time also. If you want, you can take it over and give it a new name, for example

Commit:
Merge pull request #2 from lezsakdomi/master

Some updates

rrthomas commented:
Thanks for commenting, @Kamil93. Is there some way for me to continue with the same npm name? It's the "natural" name, and would be better for existing users.

ghost commented:
I would prefer to keep original name. But here some new ideas: PcreReg, PcreV8, RegPcre, Pcre3.

rrthomas commented:
@Kamil93, since you want to keep the original name, why not accept a PR and update the module?

ghost commented:
I have no time to maintain it right now, but you can copy all source and use it for your new module.

Make a pull request in original repo and I will merge it

rrthomas commented:
Thanks, @Kamil93, I've made a PR with combined fixes from me and @lezsakdomi.

rrthomas committed:
In PCRE2.replace callbacks, pass null arguments for unmatched captures

The current design has a problem: PCRE2 only returns matches against
captures up to the last capture that matches. An optional capture may not
match, yet the overall match is still successful. This means that the number
of subpattern arguments passed to a PCRE2.replace callback can vary.

Fixing this requires adding the ability to jpcre2 to get the number of
captures in the pattern.

Use the new method getNumCaptures to set the size of the captures array, and
pad out the array with “null” values. As a result, the number of arguments
passed to the callback is always the same.

This reduces confusion, and restores sanity in TypeScript.

rrthomas committed:
Update jpcre2, and get it as a submodule

Commit:
Merge pull request #3 from rrthomas/master

Fixes from @lezsakdomi & @rrthomas (fix #1)

Commit:
Update README.md

Commit:
Update package.json

rrthomas committed:
Fix some use-after-scope errors

Commit:
Merge pull request #4 from rrthomas/master

Update jpcre2, and get it as a submodule

rrthomas committed:
Use re.replace options string as replace options

(As well as for match options.)

This means that for example the options ‘e’ and ‘E’ can be used to ignore
missing captures during replace.

rrthomas committed:
Update pcre2 to v10.36

There is no official Git repo for PCRE2, so unlike jpcre2, keep the sources
checked in normally.

rrthomas committed:
Use MatchEvaluator.nreplace, not MatchEvaluator.replace

MatchEvaluator.replace uses pcre2_substitute, which re-matches the
replacement text.

MatchEvaluator.nreplace does not, it just inserts the replacement text.

Commit:
Merge pull request #5 from rrthomas/master

Use re.replace options string as replace options, and use MatchEvaluator.nreplace to avoid nasty surprises

Commit:
Merge pull request #6 from rrthomas/pcre2-update

Update pcre2 to v10.36

rrthomas committed:
Fix ownership of strings (use-after-scope)

This fixes all instances of the problem fixed in 7d2e47b

Commit:
Merge pull request #7 from rrthomas/master

Fix ownership of strings (use-after-scope)

rrthomas committed:
Make unset captures null, not the empty string

It is important to be able to differentiate an empty match from a non-match.

Unfortunately, JPCRE2 does not permit this as-is. I have submitted a PR to
fix it: https://github.com/jpcre2/jpcre2/pull/31

Since the JPCRE2 maintainer is happy to take some form of the patch, but has
not found time to deal with it, I have replaced the JPCRE2 repo with my fork
that contains the PR.

Finally, this commit slightly simplifies binding.gyp: it removes the -O3
settings, as binding-gyp uses this by default anyway now, and updates the
language version to C++17, as this is required for my patch to JPCRE2.

Commit:
Merge pull request #8 from rrthomas/master

Make unset captures null, not the empty string

Commit:
Update package.json

Commit:
Update package.json

rrthomas committed:
lib/pcre2.js remove commented code that’s identical to existing code

The commented stanza for Symbol.match is identical to the current code for
Symbol.match, so remove it.

Commit:
Merge pull request #10 from rrthomas/master

lib/pcre2.js remove commented code that’s identical to existing code

rrthomas committed:
Switch back to jpcre2 upstream

jpcre2 upstream now has my patch to return unmatched captures as nulls, so
switch back to it.

rrthomas committed:
Improve .gitignores

Add a top-level .gitignore with patterns for Node files and for the build
directories.

Also add a couple of entries to deps/pcre2/.gitignore.
